somaticseq/run_somaticseq.py

```diff
@@ -0,0 +1,440 @@
+    parser_paired.add_argument('-normalSM',      '--normal-sample',     type=str,   help='Normal Name', default='NORMAL')
+
+    parser_paired.add_argument('-mutect',        '--mutect-vcf',        type=str,   help='MuTect VCF',        )
+    parser_paired.add_argument('-indelocator',   '--indelocator-vcf',   type=str,   help='Indelocator VCF',   )
+    parser_paired.add_argument('-mutect2',       '--mutect2-vcf',       type=str,   help='MuTect2 VCF',       )
+    parser_paired.add_argument('-varscansnv',    '--varscan-snv',       type=str,   help='VarScan2 VCF',      )
+    parser_paired.add_argument('-varscanindel',  '--varscan-indel',     type=str,   help='VarScan2 VCF',      )
+    parser_paired.add_argument('-jsm',           '--jsm-vcf',           type=str,   help='JointSNVMix2 VCF',  )
+    parser_paired.add_argument('-sniper',        '--somaticsniper-vcf', type=str,   help='SomaticSniper VCF', )
+    parser_paired.add_argument('-vardict',       '--vardict-vcf',       type=str,   help='VarDict VCF',       )
+    parser_paired.add_argument('-muse',          '--muse-vcf',          type=str,   help='MuSE VCF',          )
+    parser_paired.add_argument('-lofreqsnv',     '--lofreq-snv',        type=str,   help='LoFreq VCF',        )
+    parser_paired.add_argument('-lofreqindel',   '--lofreq-indel',      type=str,   help='LoFreq VCF',        )
+    parser_paired.add_argument('-scalpel',       '--scalpel-vcf',       type=str,   help='Scalpel VCF',       )
+    parser_paired.add_argument('-strelka-snv',   '--strelka-snv',       type=str,   help='Strelka VCF',       )
+    parser_paired.add_argument('-strelka-indel', '--strelka-indel',       type=str,   help='Strelka VCF',     )
+    parser_paired.add_argument('-tnscope',       '--tnscope-vcf',       type=str,   help='TNscope VCF',       )
+
+    parser_paired.set_defaults(which='paired')
+
+
+    # Single Sample mode
+    parser_single = sample_parsers.add_parser('single')
+    parser_single.add_argument('-bam',     '--bam-file',    type=str, help='BAM File',     required=True)
+    parser_single.add_argument('-SM',      '--sample-name', type=str, help='Sample Name',  default='TUMOR')
+    parser_single.add_argument('-mutect',  '--mutect-vcf',  type=str, help='MuTect VCF',   )
+    parser_single.add_argument('-mutect2', '--mutect2-vcf', type=str, help='MuTect2 VCF',  )
+    parser_single.add_argument('-varscan', '--varscan-vcf', type=str, help='VarScan2 VCF', )
+    parser_single.add_argument('-vardict', '--vardict-vcf', type=str, help='VarDict VCF',  )
+    parser_single.add_argument('-lofreq',  '--lofreq-vcf',  type=str, help='LoFreq VCF',   )
+    parser_single.add_argument('-scalpel', '--scalpel-vcf', type=str, help='Scalpel VCF',  )
+    parser_single.add_argument('-strelka', '--strelka-vcf', type=str, help='Strelka VCF',  )
+    parser_single.set_defaults(which='single')
+
+    args = parser.parse_args()
+
+    ##
+    inputParameters['outdir']            = args.output_directory
```

Comment:
You can use the `dest` [field](https://docs.python.org/3/library/argparse.html#dest) in argparse to have args.outdir instead of args.output_directory when you parse. That in combination with the `vars` function would convert `args` to a dictionary.

Reply:
Cool. Super useful. Didn't know that.